### PR TITLE
Add multiselect for choosing which providers to use when indexing

### PR DESCRIPTION
**Pull request by funcionante**
This is targeting the topic "Add a multi-selection component for choosing which indexers to use." on #229.

### CODE REVIEW

File: dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/management/ForceIndexing.java [hunk 1]
```diff
@@ -60,7 +60,7 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
 
     // Getting Parameters.
     String[] uris = req.getParameterValues("uri");
-    String[] pluginsName = req.getParameterValues("plugin");
+    String[] pluginsName = req.getParameterValues("plugin[]");
```
Comment (Enet4) [inline dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/management/ForceIndexing.java]: We shouldn't be doing this. Modifying the server's API is a breaking change (and in this case, an unnecessary one). We should instead adjust the web client to build multiple "plugin" query string parameters (e.g. `/foo?plugin=x&plugin=y`).

File: dicoogle/src/main/resources/webapp/js/handlers/requestHandler.js [hunk 2]
```diff
@@ -193,12 +193,13 @@ function saveIndexOptions(path, watcher, zip, saveThumbnail, effort, thumbnailSi
 
 }
 
-function forceIndex(uri){
+function forceIndex(uri, plugin){
   //console.log(state);
   // TODO use dicoogle client
   $.post(Endpoints.base + "/management/tasks/index",
```
Comment (Enet4) [inline dicoogle/src/main/resources/webapp/js/handlers/requestHandler.js]: You should instead use our Dicoogle client. The [`index`](https://bioinformatics-ua.github.io/dicoogle-client-js/interfaces/_types_dicoogle_client_d_.dicoogleclient.dicoogleaccess.html#index) method already supports a list of index providers.

**Comment by funcionante** [comment]
Now targets the whole issue #229.

File: dicoogle/src/main/resources/webapp/js/actions/pluginActions.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import Reflux from 'reflux';
+export const PluginActions = {
+  get: Reflux.createAction()
```
Comment (Enet4) [inline dicoogle/src/main/resources/webapp/js/actions/pluginActions.js]: We can export actions individually instead: `export const get = Reflux.createAction();`

File: dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js [hunk 4]
```diff
@@ -1,20 +1,37 @@
 import React from 'react';
 import {IndexStatusActions} from "../../actions/indexStatusAction";
 import {IndexStatusStore} from "../../stores/indexStatusStore";
+import {PluginActions} from "../../actions/pluginActions";
```
Comment (Enet4) [inline dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js]: With the changes in pluginActions.js, you would now do `import * as PluginActions from "../../actions/pluginActions";`

File: dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js [hunk 5]
```diff
@@ -98,6 +153,50 @@ const IndexStatusView = React.createClass({
         else{
           IndexStatusActions.stop(uid);
         }
+      },
+      _onProvidersChange: function(data) {
+        this.setState({providers: data.data});
+      },
+      _onStoragePluginsChange: function (data) {
+        var schemeList = data.data.map(plugin => plugin.scheme);
```
Comment (Enet4) [inline dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js]: Let's avoid using `var`: `let schemeList = ...`

File: dicoogle/src/main/resources/webapp/js/stores/pluginStore.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        self.trigger({
+          success: false,
+          status: xhr.status
+        });
+      }
+    );
+  }
+});
+
+export {PluginStore};
```
Comment (Enet4) [inline dicoogle/src/main/resources/webapp/js/stores/pluginStore.js]: Let's make this one a default export: `export default PluginStore;`

Then you would import it like this: `import PluginStore from ...;`

File: dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js [hunk 7]
```diff
@@ -28,20 +45,22 @@ const IndexStatusView = React.createClass({
          //Stop refresh interval
          clearInterval(refreshIntervalId);
 
+         this.unsubscribeProviders;
+         this.unsubscribeIndexStatus;
+         this.unsubscribePlugin;
```
Comment (Enet4) [inline dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js]: Oops! Parentheses are missing. It should be `this.unsubscribeProviders();` and the same to the other two lines.

Comment (funcionante) [reply dicoogle/src/main/resources/webapp/js/components/indexer/IndexStatusView.js]: Oops x) Thank you!

**Comment by bastiao** [comment]
Indeed, quite nice improvement in UI! 👍 
Just did the merge and quick fixes during merge - and will merge to dev after the Travis-CI builds.

**Pull request merged**